core/btree.rs

```diff
@@ -152,12 +152,12 @@ impl Cursor for BTreeCursor {
         Ok(())
     }
 
-    fn rowid(&self) -> Result<Ref<Option<u64>>> {
-        Ok(self.rowid.borrow())
+    fn rowid(&self) -> Result<Option<u64>> {
+        Ok(*self.rowid.borrow())
     }
```

Comment:
What is the reason for dropping Ref here? Wont it cause additional memory copies in the fast-path?

Reply:
It's more of a hack because I can't figure out how to return a ref for PseudoCursor 

```
    fn rowid(&self) -> Result<Ref<Option<u64>>> {
        let x = self
            .current
            .borrow()
            .as_ref()
            .map(|record| match record.values[0] {
                OwnedValue::Integer(rowid) => rowid as u64,
                _ => panic!("Expected integer value"),
            });
        Ok(x)
    }
```

Reply:
@benclmnt I am not sure I understand what the problem is. A `PseudoCursor` should be in `ProgramState::cursors` like BTree cursors.

Reply:
On discord, we discussed about it more. I've since changed the implementation of `Sorter::record` to follow the same function sig.